Source/CesiumRuntime/Private/CesiumGltfPointsSceneProxyUpdater.cpp

```diff
@@ -0,0 +1,86 @@
+#include "CesiumGltfPointsSceneProxyUpdater.h"
+
+#include "Cesium3DTileset.h"
+#include "CesiumGltfPointsComponent.h"
+#include "CesiumGltfPointsSceneProxy.h"
+#include "Misc/ScopeLock.h"
+#include "Misc/ScopeTryLock.h"
+
+FCesiumGltfPointsSceneProxyTilesetData::FCesiumGltfPointsSceneProxyTilesetData()
+    : PointCloudShading(),
+      MaximumScreenSpaceError(0.0),
+      UsesAdditiveRefinement(false),
+      GeometricError(0.0f),
+      Dimensions() {}
+
+void FCesiumGltfPointsSceneProxyTilesetData::UpdateFromComponent(
+    UCesiumGltfPointsComponent* Component) {
+  ACesium3DTileset* Tileset = Component->pTilesetActor;
+  PointCloudShading = Tileset->GetPointCloudShading();
+  MaximumScreenSpaceError = Tileset->MaximumScreenSpaceError;
+  UsesAdditiveRefinement = Component->UsesAdditiveRefinement;
+  GeometricError = Component->GeometricError;
+  Dimensions = Component->Dimensions;
+}
+
+FCesiumGltfPointsSceneProxyUpdater::FCesiumGltfPointsSceneProxyUpdater(
+    ACesium3DTileset* Tileset)
+    : Owner(Tileset), bNeedsUpdate(false) {}
+
+FCesiumGltfPointsSceneProxyUpdater::~FCesiumGltfPointsSceneProxyUpdater() {}
+
+void FCesiumGltfPointsSceneProxyUpdater::MarkForUpdateNextFrame() {
+  bNeedsUpdate = true;
+}
+
+void FCesiumGltfPointsSceneProxyUpdater::UpdateSettingsInProxies() {
+  if (IsValid(this->Owner) && IsInGameThread()) {
+    TransferSettingsToProxies();
+    bNeedsUpdate = false;
+  }
+}
+
+void FCesiumGltfPointsSceneProxyUpdater::UpdateSettingsInProxiesIfNeeded() {
+  if (bNeedsUpdate) {
+    UpdateSettingsInProxies();
+  }
+}
+
+void FCesiumGltfPointsSceneProxyUpdater::TransferSettingsToProxies() {
+  TInlineComponentArray<UActorComponent*> ComponentArray(Owner, true);
```

Comment:
This constructor calls `Owner->GetComponents()` itself, which is redundant with the call below.